lib/logging/lib/logging.c

```diff
@@ -0,0 +1,135 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2015-2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <syslog.h>
+
+#include <hse_util/event_counter.h>
+#include <hse_util/minmax.h>
+#include <hse_util/mutex.h>
+#include <hse_ikvdb/hse_gparams.h>
+
+#include <logging/logging.h>
+#include <logging/types.h>
+
+_Thread_local char scratch_buffer[4096];
```

Comment:
Seems a bit excessive for our purpose...

Reply:
And call it hse_print_buffer or something so that when I'm looking through the dump I know it's ours...

Reply:
Should do that for all globals...

Reply:
I didn't really see a reason to create a new buffer on the stack on every function call when it is going to be tied to each thread anyways.

But if it really makes no difference, then I can definitely just create a 4K buffer on the stack instead.

Reply:
Huh?  Leave it thread-local, but 4K seems unnecessary, maybe 1K?

Reply:
You can also make it static and move it into the function where it's used o limit its scope...

Reply:
Yea, forgot to put static. Good point about function scope.

> Should do that for all globals...

I don't see the point in this if they are static.

Reply:
Thread local variables have to be global according to the compiler error I just got.

Reply:
The point is so that I can easily identify all our symbols when I'm using objdump and similar tools...
I think you want "thread_local".
We use "static thread_local..." all over the place..